pyccl/cosmology.py

```diff
@@ -234,6 +240,22 @@ def __init__(
         if "emu" not in extra_parameters:
             extra_parameters["emu"] = {"neutrinos": "strict"}
 
+        # initialise linear Pk emulators if needed
+        self.lin_pk_emu = None
+        
+        if type(transfer_function) == emulators.BaccoemuLinear:
+            self.lin_pk_emu = transfer_function
+            transfer_function = 'baccoemu_linear'
+        
+        # initialise nonlinear Pk emulators if needed
+        self.nl_pk_emu = None
+        if type(matter_power_spectrum) == emulators.BaccoemuNonlinear:
+            self.nl_pk_emu = matter_power_spectrum
+            matter_power_spectrum = 'baccoemu_nonlinear'
+        elif type(matter_power_spectrum) == emulators.EE2Nonlinear:
+            self.nl_pk_emu = matter_power_spectrum
+            matter_power_spectrum = 'ee2_nonlinear'
+
```

Comment:
As before, this could just be
```
if type(matter_power_spectrum) == emulators.EmulatorPk:
    self.lin_pk_emu = matter_power_spectrum
    matter_power_spectrum = 'emulator'
```

Reply:
Updated also here